tableauserverclient/server/endpoint/auth_endpoint.py

```diff
@@ -28,7 +28,18 @@ def baseurl(self):
     def sign_in(self, auth_req):
         url = "{0}/{1}".format(self.baseurl, "signin")
         signin_req = RequestFactory.Auth.signin_req(auth_req)
-        server_response = self.parent_srv.session.post(url, data=signin_req, **self.parent_srv.http_options)
+        server_response = self.parent_srv.session.post(
+            url, data=signin_req, **self.parent_srv.http_options, allow_redirects=False
+        )
+        # manually handle a redirect so that we send the correct POST request instead of GET
+        # this will make e.g http://online.tableau.com work to redirect to http://east.online.tableau.com
+        if server_response.status_code == 301:
+            server_response = self.parent_srv.session.post(
+                server_response.headers["Location"],
+                data=signin_req,
+                **self.parent_srv.http_options,
+                allow_redirects=False,
```

Comment:
Any chance *this* `post` could redirect too?

Reply:
I think it could, but I don't know how many is reasonable and I don't want to set up an infinite loop.  Thoughts?

Reply:
As a datapoint, .NET defaults to 50. https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.maxautomaticredirections?view=net-6.0

Reply:
That seems like a lot! This question gave a reasonable scenario of four redirections, and it looks like browsers allow 10-20, which is closer to what I thought. I think 20 seems generous but not too much of a time-waster.

Reply:
logged for fixing in another change: https://github.com/tableau/server-client-python/issues/1127